tests/testing_common.py

```diff
@@ -561,6 +562,48 @@ def _test_merge_layers(self, model_id, config_cls, config_kwargs):
         logits_merged_from_pretrained = model_from_pretrained(**dummy_input)[0]
         self.assertTrue(torch.allclose(logits_merged, logits_merged_from_pretrained, atol=atol, rtol=rtol))
 
+    def _test_merge_layers_multi(self, model_id, config_cls, config_kwargs):
+        if config_cls not in (LoraConfig, IA3Config):
+            # Merge layers only supported for LoRA and IA³
+            return
+        if ("gpt2" in model_id.lower()) and (config_cls != LoraConfig):
+            self.skipTest("Merging GPT2 adapters not supported for IA³ (yet)")
+
+        model = self.transformers_class.from_pretrained(model_id)
+        config = config_cls(
+            base_model_name_or_path=model_id,
+            **config_kwargs,
+        )
+        model = get_peft_model(model, config)
+        model.add_adapter("adapter-2", config)
+
+        model = model.to(self.torch_device)
+
+        model_copy = copy.deepcopy(model)
+
+        if config.peft_type not in ("IA3", "LORA"):
+            with self.assertRaises(AttributeError):
+                model = model.merge_and_unload()
+
+        dummy_input = self.prepare_inputs_for_testing()
+        model.eval()
+
+        with torch.no_grad():
+            logits_all_adapters = model(**dummy_input)[0]
+
+        model_merged_1 = model.merge_and_unload(adapter_names=["adapter-2"])
+
+        with torch.no_grad():
+            logits_merged_1 = model_merged_1(**dummy_input)[0]
+
+        model_fully_merged = model_copy.merge_and_unload()
+
+        with torch.no_grad():
+            logits_fully_merged = model_fully_merged(**dummy_input)[0]
+
+        self.assertTrue(torch.allclose(logits_all_adapters, logits_fully_merged, atol=1e-4, rtol=1e-4))
+        self.assertFalse(torch.allclose(logits_all_adapters, logits_merged_1, atol=1e-4, rtol=1e-4))
```

Comment:
I'm not sure if the test is rigorous enough. Just because `model_merged_1` is different from the others is not sufficient to prove that adapter2 was merged. How about this process:

1. create PEFT model with adapter1 (default)
2. generate logits1 with adapter1
3. add adapter2
4. activate adapter2 (without merge)
5. generate logits2 with adapter2
6. check that logits1 are different from logits2
7. activate adapter1 again
8. check that logits are again like logits1
9. `merge_and_unload` adapter2
10. the unloaded model should produce logits like logits2

Furthermore, it would be good to have a similar test, only that at step 9, `merge_and_unload` is called on both adapter1 and adapter2 (so one active and one inactive adapter).